Critter.cpp

```diff
@@ -66,178 +69,171 @@ int Critter::getAge()
- **              move will not result in the critter moving out of bounds.
- **              It returns false if the move does not go out of bounds, and
- **              true if the move would result in going out of bounds.
- ******************************************************************************/
+** Critter::checkBounds(int, int, int, int)
+** Description: checkBounds is a boolean function that checks to ensure the  
+**              x and y coordinates input into the function are on the board.
+**              It returns true if the x and y coordinates are legal, and 
+**              false otherwise.
+******************************************************************************/
 bool Critter::checkBounds(int new_x, int new_y, int size_x, int size_y)
 {
     bool outOfBounce = true;
 
-    if(new_x < 0 || new_x > size_x - 1) 
+    if(new_x < 0 || new_x > size_x - 1)
       outOfBounce = false;
 
     if(new_y < 0 || new_y > size_y - 1)
       outOfBounce = false;
+
     return outOfBounce;
 }
 
+
+/******************************************************************************
+** Critter::resetMoveSuccess()
+** Description: resetMoveSuccess resets the moveSuccess to 0.  
+******************************************************************************/
 void Critter::resetMoveSuccess()
 {
    this->moveSuccess = 0;
 }
 
-int Critter::getMoveSuccess()
-{
-   return this->moveSuccess;
-}
 
-/******************************************************************
- ** Description: void breed(Critter***))
- ** PUBLIC::2019.02.05::HD-JA**
- ** function accepts 2D Critter board, identifies if critter is of age
- ** to breed, identifies whether there are available spots to breed to,
- ** then randomly selects a spot to breed to
- ******************************************************************/
+/******************************************************************************
+** Critter::breed(Critter***, int, int)
+** Description: breed is a void function that takes a 2D Critter board, the x
+**              and y coordinates of a Critter. The function checks that the
+**              Critter is of breeding age. If the Critter is of breeding age,
+**              it checks if there are any open adjacent spots to the Critter,
+**              and adds those adjacent spots to a vector where one is randomly 
+**              choosen from this vector. A new Critter is then created and 
+**              placed in the random adjacent spot.  The age of the Critter is
+**              then reset.  If there are no open spots or the Critter is not
+**              of breeding age, no breeding occurs.
+******************************************************************************/
 void Critter::breed(Critter *** boardIn, int size_x, int size_y)
 {
-    // check if critter is of breeding age
-    if ((age != 0) && (age % breedAge == 0))
-    {
+    //Check if critter is of breeding age
```

Comment:
have to make sure any changes to the code do not affect functionality/create bugs

Reply:
I changed this portion because if an Ant/Doodlebug couldn't move when it's age is breedAge, then it should be able to try and breed the next time step, but with the way it's written in the previous form, the Ant/Doodlebug would have to wait until 2*breedAge to attempt to breed again.